ddtrace/ddtrace.go

```diff
@@ -79,6 +79,48 @@ type SpanContext interface {
 	ForeachBaggageItem(handler func(k, v string) bool)
 }
 
+// ReadableSpan specifies methods to read from a span.
+type ReadableSpan interface {
+	// GetName returns the span's operation name.
+	GetName() string
+
+	// GetService returns the span's service name.
+	GetService() string
+
+	// GetResource returns the span's resource name.
+	GetResource() string
+
+	// GetType returns the span's type.
+	GetType() string
+
+	// GetDuration returns the span's duration.
+	GetDuration() int64
+
+	// Tag returns the tag value held by the given key.
+	Tag(key string) interface{}
```

Comment:
I think that ultimately it's just `Tag` and `IsError` that we need to start with IMHO

Reply:
The ReadWriteSpan interface now only contains `Span` (interface), `Tag` and `IsError`.